fix(curriculum): external fibonacci word link in CIP (#46435)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/10-coding-interview-prep/rosetta-code/fibonacci-word.md

# --description--

The Fibonacci Word may be created in a manner analogous to the Fibonacci Sequence, but it focus on iterating concatenation, with terms different each time.

<pre>Define  F_Word<sub>1</sub>  as  <strong>1</strong>
Define  F_Word<sub>2</sub>  as  <strong>0</strong>
Form   F_Word<sub>3</sub>  as  F_Word<sub>2</sub>   concatenated with  F_Word<sub>1</sub>   i.e.:  <strong>01</strong>

> Form   F_Word<sub>term</sub>  as  F_Word<sub>term-1</sub>  concatenated with  F_word<sub>term-2</sub>

Reviewer: I think the use of n is a bit more typical and consistent with the Fibonacci numbers themselves

</pre>

# --instructions--

> Write a function to return the Fibonacci Words up to `terms`. `terms` will be provided as a parameter to the function. The function should return an array of objects. The objects should be of the form: `{ term: 1, Length: 1, Entropy: 0, Word: '1' }`. `Entropy` is computed for the string `Word` and rounded to 8 decimal digits of accuracy.

Reviewer: Is this reasonably doable without providing one of the many definitions of Entropy for the learners to use?

Author: I will try to solve it tomorrow morning and update it accordingly.👍

# --description--

> The Fibonacci Word may be created in a manner analogous to the Fibonacci Sequence, but it focus on iterating concatenation.

Reviewer: woops, one last one
```suggestion
The Fibonacci Word Sequence may be created in a manner analogous to the Fibonacci Sequence, but it focuses on iterating concatenation.
```

> The Fibonacci Word may be created in a manner analogous to the Fibonacci Sequence, but it focus on iterating concatenation, with terms different `n` each time.

Reviewer: ```suggestion
The Fibonacci Word may be created in a manner analogous to the Fibonacci Sequence, but it focus on iterating concatenation.
```

<pre>Define  F_Word<sub>1</sub>  as  <strong>1</strong>
Define  F_Word<sub>2</sub>  as  <strong>0</strong>
Form   F_Word<sub>3</sub>  as  F_Word<sub>2</sub>   concatenated with  F_Word<sub>1</sub>   i.e.:  <strong>01</strong>
Form   F_Word<sub>n</sub>  as  F_Word<sub>n-1</sub>  concatenated with  F_word<sub>Term-2</sub>
</pre>

> Entropy calculation is required in this challenge, [Here is a rosetta challenge for calculating entropy](https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/entropy)

Reviewer: ```suggestion
Entropy calculation is required in this challenge, [as shown in this Rosetta Code challenge](https://www.freecodecamp.org/learn/coding-interview-prep/rosetta-code/entropy)
```

# --instructions--

> Write a function to return the Fibonacci Words up to `terms`. `terms` will be provided as a parameter `n` to the function. The function should return an array of objects. The objects should be of the form: `{ Term: 1, Length: 1, Entropy: 0, Word: '1' }`. `Entropy` is computed for the string `Word` and rounded to 8 decimal digits of accuracy.

Reviewer: ```suggestion
Write a function to return the first `n` Fibonacci Words. The number of terms is provided as a parameter to the function. The function should return an array of objects. The objects should be of the form: `{ Term: 1, Length: 1, Entropy: 0, Word: '1' }`. `Entropy` is computed for the string `Word` and rounded to 8 decimal digits of accuracy. Note that the indices of this sequence start at `1`.
```

assert(Array.isArray(fibWord(5)));
```

> `fibWord(5)` should return `[{ term:1, Length:1, Entropy:0, Word:"1" },{ term:2, Length:1, Entropy:0, Word:"0" },{ term:3, Length:2, Entropy:1, Word:"01" },{ term:4, Length:3, Entropy:0.91829583, Word:"010" },{ term:5, Length:5, Entropy:0.97095059, Word:"01001" }]`.

Reviewer: Changing
term -> Term 
would match the other fields in the output

```js
const words5 = [
  { Term: 1, Length: 1, Entropy: 0, Word: '1' },
  { Term: 2, Length: 1, Entropy: 0, Word: '0' },
  { Term: 3, Length: 2, Entropy: 1, Word: '01' },
  { Term: 4, Length: 3, Entropy: 0.91829583, Word: '010' },
  { Term: 5, Length: 5, Entropy: 0.97095059, Word: '01001' }
];

const words7 = [
  { Term: 1, Length: 1, Entropy: 0, Word: '1' },
  { Term: 2, Length: 1, Entropy: 0, Word: '0' },
  { Term: 3, Length: 2, Entropy: 1, Word: '01' },
  { Term: 4, Length: 3, Entropy: 0.91829583, Word: '010' },
  { Term: 5, Length: 5, Entropy: 0.97095059, Word: '01001' },
  { Term: 6, Length: 8, Entropy: 0.954434, Word: '01001010' },
  { Term: 7, Length: 13, Entropy: 0.9612366, Word: '0100101001001' }
];
```

## --seed-contents--

```js

> function fibWord(terms) {

Reviewer: Text above says the argument will be `n`
